src/resmom/linux/mom_start.c

```diff
@@ -2081,6 +2081,38 @@ set_job(job *pjob, struct startjob_rtn *sjr)
 					 */
 			sjr->sj_reservation = 0;
 		}
+		
+		/*
+		 * If we have a valid reservation id, save it off in case
+		 * the ALPS reservation information isn't part of the job struct
+		 * (e.g. job got killed before ALPS reservation ID was noted)
+		 * Then we can use this to cancel the ALPS reservation so we
+		 * don't leave orphaned ALPS reservations around.
+		 */
+		if(pjob->ji_extended.ji_ext.ji_reservation > 0) {
+			FILE	*fp = NULL;
+			char	filename[MAXPATHLEN];
+			int	rc;
+
+			if (rc = snprintf(filename, MAXPATHLEN, "%s/aux/%s.alpsresv", 
+				pbs_conf.pbs_home_path, pjob->ji_qs.ji_jobid) < MAXPATHLEN) {
```

Comment:
'<' has higher precedence than '=', so this statement assigns a boolean value (0 or 1) to 'rc', not the return value of snprintf().